src/coreclr/vm/codeman.h

```diff
@@ -2895,7 +2918,19 @@ class EECodeInfo
 
 #ifdef FEATURE_EH_FUNCLETS
     PTR_RUNTIME_FUNCTION GetFunctionEntry();
-    BOOL        IsFunclet()     { WRAPPER_NO_CONTRACT; return GetJitManager()->IsFunclet(this); }
+    BOOL        IsFunclet()
+    {
+        WRAPPER_NO_CONTRACT;
+        if (m_isFuncletCache == IsFuncletCache::NotSet)
```

Comment:
This looks very similar to the lazy initialization used for `GetFunctionEntry()` that is also initialized eagerly in some cases and lazily in other cases. It would be nice if the two look exactly the same:
- Move the implementation of `IsFunclet`  to .cpp file next to `GetFunctionEntry`
- Rename method on the code manager interface to `LazyIsFunclet`